### DISCUSSION TITLE
Why do you set a default logging handler?

### DISCUSSION THREAD

**Discussion by jloehel** [MISP/PyMISP#717] [Q&A] [answered]
https://github.com/MISP/PyMISP/blob/aee6945e95c647cf01d04fdf1282b6887744e041/pymisp/__init__.py#L6-L13

It's some kind of confusing if I configure logging on CLI level. [Python](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library) does not recommend to use any handler other than a NullHandler for libraries. Is there any special reason for it?

**Comment by Rafiot** [comment, marked as answer]
I don't remember the reason, but it was 3 years ago. I guess we can remove it.

**Reply by Rafiot** [reply]
And the handler is gone: https://github.com/MISP/PyMISP/commit/00ba313eae23e5d1b6842191d62dd4b56014e5e7

**Reply by chrisinmtown** [reply]
Good that you removed the handler.  Less good that you removed the call to setLevel.

**Reply by jloehel** [reply]
@chrisinmtown Why is the missing `logger.setLevel(logging.WARNING)` a issue? You can set the level easily when you setup a handler with:

~~~Python
In [1]: import logging

In [2]: logger = logging.getLogger("pymisp")

In [3]: logger.addHandler(logging.StreamHandler())

In [4]: logger.setLevel(logging.DEBUG)

In [5]: import pymisp
-c:1: DeprecationWarning: 
As our baseline system is the latest Ubuntu LTS, and Ubuntu LTS 20.04 has Python 3.8 available,
we will officially deprecate python versions below 3.8 on January 1st 2022.
**Please update your codebase.**
pymisp loaded properly
~~~

@Rafiot Maybe it's good to add a NullHandler. Something like that to `pymisp/__init__.py`:
~~~Python
# Set default logging handler to avoid "No handler found" warnings.
import logging
from logging import NullHandler

logging.getLogger(__name__).addHandler(NullHandler())
~~~

**Reply by Rafiot** [reply]
I will add the null handler, yes.
